### ISSUE TITLE
2.4.2 Page Titled (RE: Error spotted in WCAG2ICT)

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#627]
There are two related WCAG2ICT mail threads threads on this topic: 

1. [Initial email from Gregg with responses from Mary Jo and Mitch](https://lists.w3.org/Archives/Public/public-wcag2ict-tf/2025Apr/0000.html) (Initial email from Gregg quoted below). Use link to read others' added thoughts.
>I think what we have is an ERROR in WCAG2ICT for 2.4.2  Page titled
>
>for software this should be “does not apply”   with a note
>
>NOTE: Although the name of a software product could be a sufficient title if it describes the topic or purpose, software names are trademarked and trademark names cannot by law be descriptive names. It is not practical to make software names both unique and descriptive.
>
>OR we should change it to 
>
>In windowing environments - the WINDOW should have a meaningful  TITLE or NAME
>
>Note: “Name” refers to the programmatically determinable attribute “name” for the screen or window
>
>or some such...What we currently have does not make sense.  If someone does think it does  — please explain how to conform to it for Excel, or Skype or Shazam on an iPhone or Android.
3. [Follow-up email thread](https://lists.w3.org/Archives/Public/public-wcag2ict-tf/2025Apr/0013.html) regarding Bruce's first [PR 624](https://github.com/w3c/wcag2ict/pull/624/files) proposal to clarify our note. 
>From Bruce: 
>I took the liberty to draft a PR to address the issue Gregg raised on the list (link in agenda) that, while software product names are an analog to web page titles and provide descriptive identification, a good argument can be made that product names do not not describe topic or purpose.  Perhaps I should have made a GitHub Issue first, but we have already had ample discussion on the topic.
4. [Gregg's comment on PR 624](https://github.com/w3c/wcag2ict/pull/624#issuecomment-2788210173), indicating the proposal doesn't work.
5. Minutes from the discussion of list items 1 and 2 above from the [10 April meeting minutes].(https://www.w3.org/2025/04/10-wcag2ict-minutes.html#d570)
6. [Email from from Bruce](https://lists.w3.org/Archives/Public/public-wcag2ict-tf/2025Apr/0018.html) (quoted below) and alternate proposal in PR 626 created after the 10 April meeting.>On the call last week there seemed to be support for saying that 2.4.2 Page Title was not applicable to Non-web software.  I took the liberty to draft a PR for what that might look like. See [PR 626](https://github.com/w3c/wcag2ict/pull/626/files).

**Comment by GreggVan** [comment]
To make my argument more succinctly 


- the requirement is that "**the name must describe the topic or purpose”**
- If you are saying “**I think any product name “describes the topic or purpose”  no matter what the name is  (even if it is an arbitrary string of characters like  Zytmk”**
    - THEN you are saying  **“the name = a description of the topic or purpose"**
    - And then the requirement becomes  “ **The name must be**-**the name**” 
- this makes no sense as a requirement


ALSO  I really don't think  ‘Zytmk’  is a description of anything.  

- It is a name but not a description of topic or purpose.  
- If you think it is, please reply to this with a comment that tells me in other words what the topic or purpose it describes is.

**Comment by mapluke** [comment]
@GreggVan has captured the problem that we have encountered with this one in working on EN 301 549. We try to ensure that in all our requirements, and in the tests of show that you meet the requirement, all terms used in the requirement and test are clearly and unambiguously defined. In this case we have three terms, none of which is defined in WCAG:
- title: it is very clear what this is for a webpage, but for software that may present its views and the identity of the app within the operating system in different ways, there appears to be no common view of what "title" exactly refers to (which element is considered to be the title of an app, and how that might differ significantly between platforms)? There were suggestions about adding a note to explain what might be considered to be a title, but it is far from a clearly defined term.
- topic: This has a dictionary definition, such as a "subject or theme". It is hard to argue that ‘Zytmk’ meets that dictionary definition!
- purpose: This also has a dictionary definition of "the reason for which anything is done, created, or exists". It is hard to argue that ‘Zytmk’ contributes anything towards meeting this definition.

So, if we assume that for most purposes the product name is the only "title" that most retail software has, such a title cannot possibly meet even the most lax definitions of "topic" or "purpose". All software should fail this requirement, but some evaluators may pass the software because they choose to ignore the words that say it shall indicate the "topic" or "purpose". 

It goes against everything that we have tried to do in EN 301 549 to include a requirement that should always fail, but might sometimes be wrongly judged to pass! Neither outcome does anything to further the cause of accessibility!

**Comment by mitchellevan** [comment]
I would support adding another note in WCAG2ICT further explaining, how 2.4.2 Page Titled can be applied to non-web software. I do not support a statement saying that 2.4.2 does not apply to non-web software.

Here are my reasons.

### Made-up names can describe topic or purpose.

I'll focus this part of my commentary on web pages, thinking especially of a single-page website or web application. This is important because this part of the discussion is not actually limited to non-web software; it applies similarly to some web pages.

@mapluke wrote:

> topic: This has a dictionary definition, such as a "subject or theme". It is hard to argue that ‘Zytmk’ meets that dictionary definition!

Actually I do argue that "Zytmk" could be the subject or theme of a web page, if "Zytmk" were a proper noun referring to a real-world entity. Try this for comparison: Does the following web page pass or fail 2.4.2? [Zyxel - Wikipedia](https://en.wikipedia.org/wiki/Zyxel). Most people would not know what "Zyxel" is before visiting the page, but it is the actual topic of the page, so it passes 2.4.2. (Just "Zyxel" would pass too. Adding " - Wikipedia" is a best practice but not required for a pass.)

[Works of authorship](https://legalclarity.org/what-are-works-of-authorship-and-how-are-they-protected/), such as books, movies, and games, also have proper nouns as their names. A web page _about_ a work can pass 2.4.2 by using the work's made-up name as the web page title, even if few people have heard of that name before visiting the page.

Now we come to a potentially more controversial question. When a web page itself _is_ the work of authorship, can it pass 2.4.2 by using its own made-up name as its title? As @GreggVan put it:

> And then the requirement becomes “The name must be-the name”
> this makes no sense as a requirement

I'm not saying the name must be the name. I am saying that a made-up name may be used as the title of a single-page web application (SPA) or non-web software, if it's the true name of the work of authorship.

In practice this means most non-web software passes 2.4.2. This is not a theoretical claim, but one based on experience. Since the Section 508 refresh in 2017, I've been a member of auditing teams applying this manner of evaluation consistently to non-web software.

Failures, on the other hand, are somewhat theoretical because they're rare. See pass/fail examples in my [comment 17762 for ETSI Labs issue #275](https://labs.etsi.org/rep/HF/en301549/-/issues/275#note_17762). This requires human judgment, as all evaluations of 2.4.2 do.

### A title for non-web software is technically feasible across platforms.

HTML has `<title>` while non-web platforms vary, so this question is specific to non-web software. Given the word substitution "non-web software has a title", it this a feasible requirement across platforms?

In the past I had some doubts whether it's feasible, because some platforms might not have a _programmatic_ title. But after @maryjom recently pointed out [normative WCAG does not say titles are programmatic](https://lists.w3.org/Archives/Public/public-wcag2ict-tf/2025Mar/0018.html), I no longer see this as an impediment to applying 2.4.2 to non-web software.

This means evaluators must decide what counts as a title and what does not. This requires additional human judgment for non-web software, but it's no harder than, for example, deciding what counts as a 'label'. It should be based on conventions for the platform.

(The situation is different for some closed functionality software, and [WCAG2ICT has addressed a scenario for closed functionality](https://www.w3.org/TR/wcag2ict-22/#success-criteria-problematic-for-closed-functionality).)

**Comment by mitchellevan** [comment]
Meta-comment: Why does this issue matter?

It matters very little for end users. If 2.4.2 ends up void for non-web software, then it's never a failure. Likewise, evaluating 2.4.2 for non-web software will almost never be a failure, by following a testing method that many evaluators have been using for US Section 508.

The issue matters more for international standards alignment, because up till now EN 301 549 has diverged from US Section 508 on this one. Authors of the EN 301 549 revision have indicated in [ETSI Labs issue #275](https://labs.etsi.org/rep/HF/en301549/-/issues/275) that alignment with WCAG2ICT could be a deciding factor.

If Section 508 got it wrong for 2.4.2, then continuing the divergence could be the better choice — but I've explained above why I do believe WCAG2ICT (2013) and Section 508 did get it right.

**Comment by bruce-usab** [comment]
> ...And if you ask them why - it would be "well, because otherwise nothing would pass".

I think that is legit.  I almost always (usually silently) apply [argumentum ad consequentiam](https://en.wikipedia.org/wiki/Appeal_to_consequences) when 508 provision analysis takes a bad turn.

> 1. If any name automatically meets this -- then what is the purpose of the provision?  It simply requires that every product have a name.  That is a waste of space

IMHO, that's fine.  It is not necessary that every success criteria have a purpose when applied to non-web content.

> 2. Names do not meet the requirement as written.  Most names do not "describe the topic or purpose" of the software.

Gregg has an [email to the list](https://lists.w3.org/Archives/Public/public-wcag2ict-tf/2025Apr/0004.html) that includes a list of software product names that are clearly not descriptive and plainly do not reflect the purpose or function of the software.

[I did not find the list of product names persuasive.](https://lists.w3.org/Archives/Public/public-wcag2ict-tf/2025Apr/0009.html) Maybe it boils down to the difference between adjective and verb?  I think it is okay that only a small audience finds “Lotus 1-2-3” to describe the purpose of an application.

I am also fine with simply saying [2.4.2 is not applicable to software](https://github.com/w3c/wcag2ict/pull/626) and have drafted a PR (which needs work) to that effect.

**Comment by mitchellevan** [comment]
Don't wait for me, because I won't have time to attend. I will accept whichever decision the rest of the Task Force reaches about 2.4.2 for non-web software.

If the Task Force does decide to recommend that 2.4.2 not apply to non-web software, then Understanding is also needed for WCAG 2 applied to web, to clarify whether a made-up name of a web application can pass 2.4.2. Example: when I go to https://gmail.com in a private browser (not logged into Google), the title is "Gmail". I would pass that, but based on the present issue discussion, it seems some others are saying that should fail.

**Comment by JJdeGroot** [comment]
The biggest benefit of making this apply, even if all names would pass, is that people would be more likely to read the note about best practice to give "views" a clear title. 

If the choice is made to make it non applicable, please consider including the rationale behind the choice and the best practice note in its place.

**Comment by ChrisLoiselle** [comment]
As AGWG is still developing the definition of view, I feel that this needs to stay as is until that outcome is permanently determined. I feel that someone referencing WCAG2Mobile and/or WCAG2ICT for this, would be an artifact of how the term is defined and implemented by WCAG 3 within AGWG's term definition use. @maryjom I'm not sure if you wanted to add anything additional here.
